sc/src/test/java/uk/ac/ebi/atlas/experimentpage/ExperimentPageContentServiceIT.java

```diff
@@ -101,13 +110,19 @@ public void getValidDownloadsJson() {
 
     @Test
     public void getValidTsnePlotDataJson() {
-        JsonObject result = this.subject.getTsnePlotDataAsJson();
+        String experimentAccession = jdbcTestUtils.fetchRandomSingleCellExperimentAccession();
+        JsonObject result = this.subject.getTsnePlotDataAsJson(experimentAccession);
 
         assertThat(result.has("suggesterEndpoint"), is(true));
         assertThat(result.get("suggesterEndpoint").getAsString(), is("json/suggestions"));
 
         assertThat(result.has("ks"), is(true));
-        assertThat(result.get("ks").getAsJsonArray().size(), is(greaterThanOrEqualTo(1)));
+        assertThat(
+                ImmutableSet.copyOf(result.get("ks").getAsJsonArray()).stream()
+                        .map(JsonElement::getAsInt)
+                        .collect(toSet()),
+                containsInAnyOrder(
+                        jdbcTestUtils.fetchKsFromCellClusters(experimentAccession).toArray(new Integer[0])));
```

Comment:
How come the test is retrieving ks from the database, whereas the `getTsnePlotDataAsJson` method is using the file directly?

Reply:
It indirectly ensures that the clusters file was properly imported to the DB. I also realised it was a bit nefarious, but if at some point it fails, it will signal that something has gone wrong. In the future we want to retrieve the valid ks from the DB as well, and store it in the experiments table (at least I think it’s the best alternative), so we’ll end up getting the ks from there instead of the file.